mono/metadata/threads.c

```diff
@@ -1947,42 +1941,36 @@ mono_thread_internal_current_handle (void)
 static MonoThreadInfoWaitRet
 mono_join_uninterrupted (MonoThreadHandle* thread_to_join, gint32 ms, MonoError *error)
 {
-	MonoException *exc;
-	MonoThreadInfoWaitRet ret;
-	gint64 start;
-	gint32 diff_ms;
 	gint32 wait = ms;
+	gint64 const start = (ms == -1) ? 0 : mono_msec_ticks ();
 
-	error_init (error);
+	// Allocate handle outside of the loop.
+	MonoExceptionHandle exc = MONO_HANDLE_NEW (MonoException, NULL);
```

Comment:
mono_join_uninterrupted should probably setup a handle frame.

Right now there's a single caller and it's an icall, but it's too easy to inadvertantly add another caller and start to leak handles.  (particularly with functions like this one where the declaration doesn't include and handles)

Reply:
We could merge mono_join_uninterrupted into its only caller..

Reply:
I see your point about the missing frame.

Reply:
I don't have a strong opinion about inlining mono_join_uninterrupted one way or the other - it seems like a useful function to have, but evidently we haven't had another use for it.

One thing that might be nice to think about (but not in this PR) is whether we can find some way to have a single mechanism for timeout math that's in this file - we do the loop and subtract `mono_msec_ticks` pattern in three or four different places all for basically the same reason.